source/brailleTables.py

```diff
@@ -82,325 +97,470 @@ def listTables():
+	# braille settings dialog.
+	addTable("ro.ctb", _("Romanian"))
+	# Translators: The name of a braille table displayed in the
+	# braille settings dialog.
+	addTable("ru.ctb", _("Russian braille for computer code"))
+	# Translators: The name of a braille table displayed in the
+	# braille settings dialog.
+	addTable("ru-ru-g1.utb", _("Russian grade 1"))
+	# Translators: The name of a braille table displayed in the
+	# braille settings dialog.
+	addTable("sa-in-g1.utb", _("Sanskrit grade 1"))
+	# Translators: The name of a braille table displayed in the
+	# braille settings dialog.
+	addTable("Se-Se.ctb", _("Swedish 8 dot computer braille"))
+	# Translators: The name of a braille table displayed in the
+	# braille settings dialog.
+	addTable("Se-Se-g1.utb", _("Swedish grade 1"))
+	# Translators: The name of a braille table displayed in the
+	# braille settings dialog.
+	addTable("sk-g1.ctb", _("Slovak grade 1"))
+	# Translators: The name of a braille table displayed in the
+	# braille settings dialog.
+	addTable("sl-si-comp8.ctb", _("Slovenian 8 dot computer braille"))
+	# Translators: The name of a braille table displayed in the
+	# braille settings dialog.
+	addTable("sl-si-g1.utb", _("Slovenian grade 1"))
+	# Translators: The name of a braille table displayed in the
+	# braille settings dialog.
+	addTable("sr-g1.ctb", _("Serbian grade 1"))
+	# Translators: The name of a braille table displayed in the
+	# braille settings dialog.
+	addTable("ta-ta-g1.ctb", _("Tamil grade 1"))
+	# Translators: The name of a braille table displayed in the
+	# braille settings dialog.
+	addTable("te-in-g1.utb", _("Telugu grade 1"))
+	# Translators: The name of a braille table displayed in the
+	# braille settings dialog.
+	addTable("tr.ctb", _("Turkish grade 1"))
+	# Translators: The name of a braille table displayed in the
+	# braille settings dialog.
+	addTable("uk.utb", _("Ukrainian"))
+	# Translators: The name of a braille table displayed in the
+	# braille settings dialog.
+	addTable("unicode-braille.utb", _("Unicode braille"), output=False)
+	# Translators: The name of a braille table displayed in the
+	# braille settings dialog.
+	addTable("vi-g1.ctb", _("Vietnamese grade 1"))
+	# Translators: The name of a braille table displayed in the
+	# braille settings dialog.
+	addTable("zhcn-g1.ctb", _("Chinese (China, Mandarin) grade 1"))
+	# Translators: The name of a braille table displayed in the
+	# braille settings dialog.
+	addTable("zhcn-g2.ctb", _("Chinese (China, Mandarin) grade 2"))
+	# Translators: The name of a braille table displayed in the
+	# braille settings dialog.
+	addTable("zh-hk.ctb", _("Chinese (Hong Kong, Cantonese)"))
+	# Translators: The name of a braille table displayed in the
+	# braille settings dialog.
+	addTable("zh-tw.ctb", _("Chinese (Taiwan, Mandarin)"))
+
+	# Add custom tables.
+	import os
+	from logHandler import log
+	customDirs = _getCustomTablesDirs()
+	tablesDirs[:0] = customDirs

```

Comment:
I do not understand this line.

Reply:
`tablesDirs[:0] = customDirs` is a "slice-replace".
A slower equivalent would be:
```
for dir_ in reversed(customDirs):
    tablesDirs.insert(0, dir_)
```
That is, it prepends `tableDirs` with the content of `customDirs` while ensuring an eventual external reference to `tablesDirs` remains valid.
Another way of writing it, but invalidating an external reference, would be:
`tablesDirs = customDirs + tablesDirs`

Reply:
Comment added as of deb3836.